rpc_jobs/standard_job_premerge.yml

```diff
@@ -0,0 +1,134 @@
+- project:
+    name: "gating"
+    repo_url: "https://github.com/rcbops/rpc-gating"
+    branches:
+      - "master"
+    image:
+      - xenial:
+          IMAGE: "Ubuntu 16.04 LTS (Xenial Xerus) (PVHVM)"
+    scenario:
+      - "functional"
+    action:
+      - "test"
+    jobs:
+      - 'PR_{name}-{image}-{scenario}-{action}'
+
+- job-template:
+    name: 'PR_{name}-{image}-{scenario}-{action}'
+    project-type: pipeline
+    concurrent: true
+    properties:
+      - build-discarder:
+          num-to-keep: "30"
+      - github:
+          url: "{repo_url}"
+    parameters:
+      - rpc_gating_params
+      - instance_params:
+          IMAGE: "{IMAGE}"
+          FLAVOR: "{FLAVOR}"
+          REGIONS: "{REGIONS}"
+          FALLBACK_REGIONS: "{FALLBACK_REGIONS}"
+      - string:
+          name: STAGES
+          default: >-
+            Allocate Resources,
+            Connect Slave,
+            Cleanup,
+            Destroy Slave
+          description: |
+            Pipeline stages to run CSV. Note that this list does not influence execution order.
+            Options:
+              Allocate Resources
+              Connect Slave
+              Cleanup
+              Destroy Slave
+    triggers:
+      - github-pull-request:
+          org-list:
+            - rcbops
+          github-hooks: true
+          trigger-phrase: '.*recheck_(cit_)?all.*|.*recheck_(cit_)?{image}_{scenario}_{action}.*'
+          only-trigger-phrase: false
+          white-list-target-branches: "{branches}"
+          auth-id: "github_account_rpc_jenkins_svc"
+          status-context: 'CIT/{image}_{scenario}_{action}'
+          cancel-builds-on-update: true
+
+    dsl: |
+      library "rpc-gating@${{RPC_GATING_BRANCH}}"
+
+      // Pass details about the job parameters through
+      // to the target environment so that scripts can
+      // use them to adapt behaviour.
+      env.RE_JOB_NAME = "{name}"
+      env.RE_JOB_IMAGE = "{image}"
+      env.RE_JOB_SCENARIO = "{scenario}"
+      env.RE_JOB_ACTION = "{action}"
+      env.RE_JOB_FLAVOR = "{FLAVOR}"
+      env.RE_JOB_TRIGGER = "PR"
+
+      // Set the default environment variables used
+      // by the artifact and test result collection.
+      env.RE_HOOK_ARTIFACT_DIR="${{env.WORKSPACE}}/artifacts"
+      env.RE_HOOK_RESULT_DIR="${{env.WORKSPACE}}/results"
```

Comment:
These must be set within runonpubcloud, as there is no workspace at this point as no node has been allocated.

Reply:
Done.